PR: https://github.com/odoo/odoo/pull/

From: 89e358caa8627c5eff21ee63015dece0997db2b4
From: eco-odoo
Date: 2018-08-10 14:08:47

Structural Changes: 6
Total Changes: 402

[FIX] onboarding: Fix/Update onboarding to v2.0

Specification
=============

- Fix lots of bug due to the BS4 migration
- Improve steps
- Validate some steps (Like the company configuration) in all the
  onboarding bars if done in one onboarding bar.
- Only animate the confetti one time if a step is done, instead of
  on each view loading

For more information, see: https://www.odoo.com/web#id=1869513&action=333&active_id=965&model=project.task&view_type=form&menu_id=4720

================================= pseudo patch: =================================

--- a/odoo/addons/base/data/report_paperformat_data.xml
+++ b/odoo/addons/base/data/report_paperformat_data.xml
@@ -2,7 +2,7 @@
 <odoo>
     <data noupdate="1">
         <record id="paperformat_euro" model="report.paperformat">
-            <field name="name">European A4</field>
+            <field name="name">A4</field>
             <field name="default" eval="True" />
             <field name="format">A4</field>
             <field name="page_height">0</field>

--- a/odoo/addons/base/models/ir_actions_report.py
+++ b/odoo/addons/base/models/ir_actions_report.py
@@ -98,7 +98,7 @@ class IrActionsReport(models.Model):
     groups_id = fields.Many2many('res.groups', 'res_groups_report_rel', 'uid', 'gid', string='Groups')
     multi = fields.Boolean(string='On Multiple Doc.', help="If set to true, the action will not be displayed on the right toolbar of a form view.")
 
-    paperformat_id = fields.Many2one('report.paperformat', 'Paper format')
+    paperformat_id = fields.Many2one('report.paperformat', 'Paper Format')
     print_report_name = fields.Char('Printed Report Name',
                                     help="This is the filename of the report going to download. Keep empty to not change the report filename. You can use a python expression with the 'object' and 'time' variables.")
     attachment_use = fields.Boolean(string='Reload from Attachment',

--- a/odoo/addons/base/models/res_company.py
+++ b/odoo/addons/base/models/res_company.py
@@ -55,7 +55,7 @@ class Company(models.Model):
     website = fields.Char(related='partner_id.website')
     vat = fields.Char(related='partner_id.vat', string="Tax ID")
     company_registry = fields.Char()
-    paperformat_id = fields.Many2one('report.paperformat', 'Paper format', default=lambda self: self.env.ref('base.paperformat_euro', raise_if_not_found=False))
+    paperformat_id = fields.Many2one('report.paperformat', 'Paper Format', default=lambda self: self.env.ref('base.paperformat_euro', raise_if_not_found=False))
     external_report_layout = fields.Selection([
         ('background', 'Background'),
         ('boxed', 'Boxed'),
@@ -66,8 +66,9 @@ class Company(models.Model):
     _sql_constraints = [
         ('name_uniq', 'unique (name)', 'The company name must be unique !')
     ]
-    base_onboarding_company_done = fields.Boolean("Onboarding company step done",
-        compute="_compute_base_onboarding_company_done")
+
+    base_onboarding_company_state = fields.Selection([
+        ('not_done', "Not done"), ('just_done', "Just done"), ('done', "Done")], string="State of the onboarding company step", default='not_done')
 
     @api.model_cr
     def init(self):
@@ -239,6 +240,7 @@ class Company(models.Model):
             currency = self.env['res.currency'].browse(values['currency_id'])
             if not currency.active:
                 currency.write({'active': True})
+
         return super(Company, self).write(values)
 
     @api.constrains('parent_id')
@@ -266,15 +268,37 @@ class Company(models.Model):
         else:
             return res
 
-    @api.depends('street')
-    def _compute_base_onboarding_company_done(self):
-        """ The company onboarding step is marked as done if street1 is filled. """
-        for record in self:
-            record.base_onboarding_company_done = bool(record.street)
-
     @api.model
     def action_open_base_onboarding_company(self):
         """ Onboarding step for company basic information. """
         action = self.env.ref('base.action_open_base_onboarding_company').read()[0]
         action['res_id'] = self.env.user.company_id.id
         return action
+
+    def set_onboarding_step_done(self, step_name):
+        if self[step_name] == 'not_done':
+            self[step_name] = 'just_done'
+
+    def get_and_update_onbarding_state(self, onboarding_state, steps_states):
+        """ Needed to display onboarding animations only one time. """
+        old_values = {}
+        all_done = True
+        for step_state in steps_states:
+            old_values[step_state] = self[step_state]
+            if self[step_state] == 'just_done':
+                self[step_state] = 'done'
+            all_done = all_done and self[step_state] == 'done'
+
+        if all_done:
+            if self[onboarding_state] == 'not_done':
+                # string `onboarding_state` instead of variable name is not an error
+                old_values['onboarding_state'] = 'just_done'
+            else:
+                old_values['onboarding_state'] = 'done'
+            self[onboarding_state] = 'done'
+        return old_values
+
+    @api.multi
+    def action_save_onboarding_company_step(self):
+        if bool(self.street):
+            self.set_onboarding_step_done('base_onboarding_company_state')

--- a/odoo/addons/base/static/src/scss/onboarding.scss
+++ b/odoo/addons/base/static/src/scss/onboarding.scss
@@ -16,23 +16,31 @@ $o-onboarding-violet-bg-2: #475DA4;
 $o-onboarding-violet-ui-1: #F39D9B;
 $o-onboarding-violet-ui-2: #FBEEC8;
 
+@mixin o-onboarding-vertical-gradient($startColor: #555, $endColor: #333) {
+    background-color: mix($startColor, $endColor);
+    background-image: -webkit-gradient(linear, left top, left bottom, from($startColor), to($endColor));
+    background-image: -webkit-linear-gradient(top, $startColor, $endColor);
+    background-image: -moz-linear-gradient(top, $startColor, $endColor);
+    background-image: -ms-linear-gradient(top, $startColor, $endColor);
+    background-image: -o-linear-gradient(top, $startColor, $endColor);
+    background-image: linear-gradient(to bottom, $startColor, $endColor);
+}
+
 .o_onboarding {
     position: relative;
     overflow: hidden;
-    background: white center;
+    background: white center top;
     background-size: cover;
-    background-position: center top;
     border-bottom: 1px solid gray('300');
 
     .o_onboarding_wrap {
         padding: $o-horizontal-padding*2 $o-horizontal-padding;
     }
-
-    .o_onboarding_folded_title {
-        display: none;
+    .o_onboarding_step_title {
+        cursor: pointer;
     }
 
-    .o_onboarding_btn_fold, .o_onboarding_btn_close {
+    .o_onboarding_btn_close {
         @include o-position-absolute(auto, 0, 0);
         padding: 6px $o-horizontal-padding + 6px;
         cursor: pointer;
@@ -40,10 +48,25 @@ $o-onboarding-violet-ui-2: #FBEEC8;
 
     .o_onboarding_btn_close {
         top: 0;
+        bottom: auto;
     }
-
-    .modal-body, .modal-title {
-        color: black;
+    .o_onboarding_completed_message {
+        @include o-position-absolute(auto, 0,0,0);
+        //transform: translateY(115%);
+        //transition: transform $o-onboarding-dot-timing;
+        box-shadow: 0 2px 10px gray(600);
+        padding: $o-horizontal-padding;
+        background-color: white;
+        color: $o-main-text-color;
+        opacity: 0;
+        z-index: -1;
+        &.o_onboarding_steps_just_done {
+            animation: o_onb_fadeIn ($o-onboarding-dot-timing*3);
+        }
+        &.o_onboarding_steps_done {
+            opacity: 1;
+            z-index: auto;
+        }
     }
 
     .o_onboarding_steps, .o_onboarding_step {
@@ -56,10 +79,6 @@ $o-onboarding-violet-ui-2: #FBEEC8;
 
         @include media-breakpoint-down(sm) {
            flex-flow: column nowrap;
-
-            .o_onboarding_folded_title {
-                border-right: none;
-            }
         }
     }
 
@@ -83,7 +102,7 @@ $o-onboarding-violet-ui-2: #FBEEC8;
 
         &:first-of-type {
             .o_onboarding_progress {
-                display: none;
+                display: none !important;
             }
 
             .o_onboarding_dot {
@@ -92,6 +111,9 @@ $o-onboarding-violet-ui-2: #FBEEC8;
         }
 
         .o_onboarding_step_content {
+            @include media-breakpoint-up(lg) {
+                min-width: 230px;
+            }
             max-width: 230px;
             padding: 0 10px;
 
@@ -166,14 +188,9 @@ $o-onboarding-violet-ui-2: #FBEEC8;
             }
 
             .o_onboarding_all_done {
-                animation: bounceIn ($o-onboarding-dot-timing*2);
                 display: block;
             }
 
-            .o_onboarding_step_content_info {
-                animation: o_onb_fadeIn ($o-onboarding-dot-timing*3);
-            }
-
             + .o_onboarding_done {
                 .o_onboarding_dot {
                     transition-delay: 0s;
@@ -188,100 +205,24 @@ $o-onboarding-violet-ui-2: #FBEEC8;
                 }
             }
         }
-    }
-
-    @keyframes o_onb_fadeIn {
-        0%  {opacity: 0}
-        33% {opacity: 0}
-        100% {opacity: 1}
-    }
-
-    // =========    Transition    =========
-    &.collapsing {
-        transition: all .25s;
-        min-height: 30px;
-    }
-
-    // =========   Folded Design    =========
-    // ======================================
-
-    &.collapse:not(.in) {
-        min-height: 30px;
-        background-image: none!important;
-        display: block;
-
-        .o_onboarding_hide_folded {
-            display: none!important;
-        }
 
-        .o_onboarding_btn_fold {
-            top: 0;
-            left: 0;
-            transform: scaleY(-1);
-        }
+        &.o_onboarding_just_done {
 
-        .o_onboarding_wrap {
-            padding: 0 $o-horizontal-padding;
-
-            .o_onboarding_steps {
-                padding: 0;
-                justify-content: baseline;
-            }
-
-            .o_onboarding_folded_title {
-                padding: 6px 10px 6px 0;
-
-                @include media-breakpoint-up(md) {
-                    border-right: 1px solid gray('300');
-                }
+            .o_onboarding_all_done {
+                animation: bounceIn ($o-onboarding-dot-timing*2);
+                display: block;
             }
-
-            .o_onboarding_folded_title, .o_onboarding_step {
-                display: inline-flex;
+            .o_onboarding_step_content_info {
+                animation: o_onb_fadeIn ($o-onboarding-dot-timing*3);
             }
+        }
 
-            .o_onboarding_step {
-                align-items: flex-start;
-                flex: 0 1 auto;
-                padding-right: 12px;
-
-                &:not(:last-of-type) {
-                    &:before, &:after {
-                        content: " ";
-                        @include o-position-absolute(0, 0);
-
-                        border-top: floor($o-statusbar-height/2) solid transparent;
-                        border-bottom: ceil($o-statusbar-height/2) solid transparent;
-                        border-right: none;
-                        border-left: $o-statusbar-arrow-width solid $o-view-background-color;
-                        -moz-transform: scale(0.9999); // Smooth the triangle on firefox
-                    }
-
-                    &:before {
-                        right: -1px;
-                        border-left-color: gray('300');
-                    }
-                }
-
-                .o_onboarding_step_content {
-                    padding: 0;
-                    display: flex;
-                }
-
-                .o_onboarding_step_content_info .o_onboarding_step_title {
-                    margin: 7px!important;
-                    font-size: $font-size-base;
-                }
-
-                .o_onboarding_all_done {
-                    margin: 6px 10px 0 -4px!important;
+    }
 
-                    > i:before {
-                        content: "\f00c";
-                    }
-                }
-            }
-        }
+    @keyframes o_onb_fadeIn {
+        0%  {opacity: 0}
+        33% {opacity: 0}
+        100% {opacity: 1}
     }
 
     // =========  Color Variations  =========
@@ -289,12 +230,12 @@ $o-onboarding-violet-ui-2: #FBEEC8;
 
     // =========  Variation Mixin  ==========
     @mixin o-onboarding-variation($bg1, $bg2, $ui1, $ui2) {
-        &, .o_onboarding_btn_fold, .o_onboarding_step_title, .o_onboarding_step_action, a.o_onboarding_all_done, .o_onboarding_all_done > .fa, .o_onboarding_btn_close {
-            color: white;
+        &, .o_onboarding_step_title, .o_onboarding_step_action, a.o_onboarding_all_done, .o_onboarding_all_done > .fa, .o_onboarding_btn_close {
+            color: white !important;
         }
 
         .o_onboarding_wrap {
-            background-image: linear-gradient(to bottom, fade-out($bg1, 0.5), $bg2);
+            @include o-onboarding-vertical-gradient(fade-out($bg1, 0.5), $bg2);
             box-shadow: inset 0 -7px 20px -5px fade-out(black, 0.7);
         }
 
@@ -324,15 +265,15 @@ $o-onboarding-violet-ui-2: #FBEEC8;
     }
 
     // =========  Variation Templates  ==========
-    &.o_onboarding_blue.collapse.in, &.o_onboarding_blue.collapsing {
+    &.o_onboarding_blue {
         @include o-onboarding-variation($o-onboarding-blue-bg-1, $o-onboarding-blue-bg-2, $o-onboarding-blue-ui-1, $o-onboarding-blue-ui-2);
     }
 
-    &.o_onboarding_orange.collapse.in, &.o_onboarding_orange.collapsing {
+    &.o_onboarding_orange {
         @include o-onboarding-variation($o-onboarding-orange-bg-1, $o-onboarding-orange-bg-2, $o-onboarding-orange-ui-1, $o-onboarding-orange-ui-2);
     }
 
-    &.o_onboarding_violet.collapse.in, &.o_onboarding_violet.collapsing {
+    &.o_onboarding_violet {
         @include o-onboarding-variation($o-onboarding-violet-bg-1, $o-onboarding-violet-bg-2, $o-onboarding-violet-ui-1, $o-onboarding-violet-ui-2);
     }
 
@@ -451,10 +392,10 @@ $o-onboarding-violet-ui-2: #FBEEC8;
         }
     }
 
-    .o_onboarding_done {
+    .o_onboarding_just_done {
         // Activate confetti cannon when step is done
         .o_onboarding_confetti {
             display: block;
         }
     }
-}
\ No newline at end of file
+}

--- a/odoo/addons/base/views/onboarding_views.xml
+++ b/odoo/addons/base/views/onboarding_views.xml
@@ -1,85 +1,89 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
     <template id="onboarding_container">
-        <div class="o_onboarding_container collapse in show">
-            <t t-call-assets="base.assets_onboarding" />
-            <div t-att-class="'o_onboarding ' + classes + (' collapse' if folded_by_default else ' collapse in show')"
+        <div class="modal o_onboarding_modal" tabindex="-1" role="dialog">
+            <div class="modal-dialog" role="document">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <h5 class="modal-title">Closing the onboarding panel</h5>
+                        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                            <span aria-label="Close">&amp;times;</span>
+                        </button>
+                    </div>
+                    <div class="modal-body">
+                        <p>Are you sure you want to close this onboarding panel?
+                                    This action is definitive.</p>
+                    </div>
+                    <div class="modal-footer">
+                        <button type="button" class="btn btn-secondary" data-dismiss="modal">No
+                        </button>
+                        <a type="action" class="btn btn-primary" data-dismiss="modal"
+                        data-toggle="collapse" href=".o_onboarding_container"
+                        t-att-data-model="close_model" t-att-data-method="close_method">
+                            Yes
+                        </a>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div class="o_onboarding_container collapse show">
+            <div t-att-class="'o_onboarding ' + ('o_onboarding_completed ' if steps_completed else ' ') + classes"
                  t-att-style="'background-image:url(' + bg_image + ')'">
                 <div class="o_onboarding_wrap">
-                    <a t-if="not no_close_btn" href="#" data-toggle="modal" data-target=".o_onboarding_modal"
-                       class="float-right o_onboarding_btn_close o_onboarding_hide_folded">
+                    <a href="#" data-toggle="modal" data-target=".o_onboarding_modal"
+                       class="float-right o_onboarding_btn_close">
                         <i class="fa fa-times" title="Close the onboarding panel" />
                     </a>
-                    <!-- Modal -->
-                    <div class="o_onboarding_modal modal fade" tabindex="-1" role="dialog" aria-labelledby="onboardingCloseLabel">
-                        <div class="modal-dialog" role="document">
-                            <div class="modal-content">
-                                <div class="modal-header">
-                                    <h4 class="modal-title" id="onboardingCloseLabel">Closing the onboarding panel</h4>
-                                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
-                                        <span aria-label="close the modal">&amp;times;</span>
-                                    </button>
-                                </div>
-                                <div class="modal-body">
-                                    Are you sure you want to close this onboarding panel?
-                                    This action is definitive.
-                                </div>
-                                <div class="modal-footer">
-                                    <button type="button" class="btn btn-secondary" data-dismiss="modal">
-                                        No
-                                    </button>
-                                    <a type="action" class="btn btn-primary" data-dismiss="modal"
-                                        data-toggle="collapse" href=".o_onboarding_container"
-                                        t-att-data-model="close_model"
-                                        t-att-data-method="close_method">
-                                       Yes
-                                    </a>
-                                </div>
-                            </div>
-                        </div>
-                    </div>
                     <div class="o_onboarding_steps">
-                        <span t-if="not no_fold_btn" class="o_onboarding_folded_title">
-                            Configuration steps:
-                        </span>
                         <t t-raw="0" />
                     </div>
-                    <a t-if="not no_fold_btn or folded_by_default" data-toggle="collapse"
-                       href=".o_onboarding" type="action" t-att-data-model="fold_model"
-                       t-att-data-method="fold_method" class="o_onboarding_btn_fold text-right">
-                        <i class="fa fa-chevron-up" title="Fold the onboarding panel" />
-                    </a>
+                    <div t-att-class="'o_onboarding_completed_message text-center' + (' o_onboarding_steps_done' if state.get('onboarding_state') in ('done', 'just_done') else '') + (' o_onboarding_steps_just_done' if state.get('onboarding_state') == 'just_done' else '')">
+                        <i class="fa fa-check text-success mr8" />
+                        <span class="h3 mr16">
+                            <t t-if="text_completed" t-esc="text_completed" />
+                            <t t-else="">Nice work!</t>
+                        </span>
+                        <div class="btn btn-primary btn-sm">
+                            <a type="action" class="btn btn-primary" data-toggle="collapse" href=".o_onboarding_container"
+                            t-att-data-model="close_model" t-att-data-method="close_method">
+                                Close
+                            </a>
+                        </div>
+                    </div>
                 </div>
             </div>
         </div>
     </template>
     <template id="onboarding_step">
-        <div t-att-class="'o_onboarding_step ' + ('o_onboarding_done' if done else 'o_onboarding_current')">
+        <div t-att-class="'o_onboarding_step ' + ('o_onboarding_done' if state in ('done', 'just_done') else 'o_onboarding_current') + (' o_onboarding_just_done' if state == 'just_done' else '')">
             <t t-call="base.onboarding_confetti" />
-            <div class="o_onboarding_progress o_onboarding_hide_folded d-none d-md-inline-block progress">
+            <div class="o_onboarding_progress d-none d-md-inline-block progress">
                 <div class="progress-bar"></div>
             </div>
-            <span class="o_onboarding_dot o_onboarding_hide_folded d-none d-md-inline-block fa fa-check" />
+            <span class="o_onboarding_dot d-none d-md-inline-block fa fa-check" />
             <div class="o_onboarding_step_content">
                 <div class="o_onboarding_step_content_info">
                     <h4 class="o_onboarding_step_title mt16">
-                        <t t-esc="title" />
+                        <a type="action" data-reload-on-close="true" role="button"
+                            t-att-data-method="method" t-att-data-model="model">
+                            <t t-esc="title" />
+                        </a>
                     </h4>
-                    <p class="o_onboarding_hide_folded mb16">
+                    <p class="mb16">
                         <t t-esc="description" />
                     </p>
                 </div>
-                <a class="o_onboarding_step_action o_onboarding_hide_folded btn" type="action"
+                <a class="o_onboarding_step_action btn" type="action"
                    data-reload-on-close="true" role="button" t-att-data-method="method"
                    t-att-data-model="model">
                     <t t-if="btn_text" t-esc="btn_text" />
                     <t t-else="">Let's do it</t>
                 </a>
-                <a class="o_onboarding_all_done mt16" t-att-name="form_ref" type="action"
+                <a class="o_onboarding_all_done mt16" type="action"
                    data-reload-on-close="true" role="button" t-att-data-method="method"
                    t-att-data-model="model">
                     <i t-attf-class="text-success mr6 fa #{done_icon if done_icon else 'fa-check'}" />
-                    <span class="o_onboarding_hide_folded">
+                    <span>
                         <t t-if="done_text" t-esc="done_text" />
                         <t t-else="">All done!</t>
                     </span>
@@ -132,22 +136,15 @@
     <template id="onboarding_company_step">
         <t t-call="base.onboarding_step">
             <t t-set="title">Company Data</t>
-            <t t-set="description">Your information is displayed on the invoices you send to
-                clients.
+            <t t-set="description">
+                Set your company's data for documents header/footer.
             </t>
             <t t-set="btn_text">Let's start!</t>
             <t t-set="method" t-value="'action_open_base_onboarding_company'" />
             <t t-set="model" t-value="'res.company'" />
-            <t t-set="done" t-value="company.base_onboarding_company_done" />
+            <t t-set="state" t-value="state.get('base_onboarding_company_state')" />
         </t>
     </template>
-    <template id="assets_onboarding">
-        <!-- Include backend mixins and variables, completed with bootstrap ones -->
-        <t t-call="web._assets_backend_helpers"/>
-        <link rel="stylesheet" type="text/scss" href="/web/static/lib/bootstrap/scss/_variables.scss"/>
-
-        <link rel="stylesheet" type="text/scss" href="/base/static/src/scss/onboarding.scss"/>
-    </template>
     <!-- COMPANY FORM -->
     <record id="base_onboarding_company_form" model="ir.ui.view">
         <field name="name">base.company.onboarding.form</field>
@@ -156,13 +153,10 @@
         <field name="mode">primary</field>
         <field name="priority">1000</field>
         <field name="arch" type="xml">
-            <field name="logo" position="attributes">
-                <attribute name="attrs">{'invisible': 1}</attribute>
-            </field>
             <xpath expr="//group[@name='social_media']" position="replace" />
             <form position="inside">
                 <footer position="replace">
-                    <button special="save" string="Save" class="btn btn-primary" type="object" />
+                    <button name="action_save_onboarding_company_step" class="btn btn-primary" type="object" string="Apply" />
                     <button special="cancel" string="Cancel" />
                 </footer>
             </form>

--- a/odoo/addons/base/views/res_company_views.xml
+++ b/odoo/addons/base/views/res_company_views.xml
@@ -187,44 +187,37 @@
             <field name="priority">2048</field>
             <field name="arch" type="xml">
                 <form class="o_company_document_layout">
-                    <group col="2">
-                        <group>
-                            <label for="external_report_layout" string="Template" colspan="2" />
-                            <field name="external_report_layout" colspan="2" nolabel="1"
-                                   class="image_selection_container"
-                                   widget="image_selection" options="{
-                                        'background': {
-                                            'image_link': '/base/static/img/preview_background.png',
-                                            'preview_link': '/base/static/pdf/preview_background.pdf'
-                                        },
-                                        'boxed': {
-                                            'image_link': '/base/static/img/preview_boxed.png',
-                                            'preview_link': '/base/static/pdf/preview_boxed.pdf'
-                                        },
-                                        'clean': {
-                                            'image_link': '/base/static/img/preview_clean.png',
-                                            'preview_link': '/base/static/pdf/preview_clean.pdf'
-                                        },
-                                        'standard': {
-                                            'image_link': '/base/static/img/preview_standard.png',
-                                            'preview_link': '/base/static/pdf/preview_standard.pdf'
-                                        }
-                                    }" />
-                        </group>
-                        <group>
-                            <label for="logo" colspan="2" string="Company logo" />
-                            <field name="logo" widget="image" nolabel="1" colspan="2" />
-                            <label for="report_header" colspan="2" string="Company Tagline" />
-                            <field name="report_header" nolabel="1" colspan="2"
-                                   placeholder="e.g. Global Business Solutions" />
-                            <label for="report_footer" colspan="2"
-                                   string="Extra information in the footer" />
-                            <field name="report_footer" nolabel="1" colspan="2"
-                                   placeholder="e.g. Opening hours, bank accounts (one per line)" />
-                        </group>
+                    <group>
+                        <field name="logo" widget="image" />
+                        <field name="external_report_layout" string="Layout"
+                               class="image_selection_container"
+                               widget="image_selection" options="{
+                                    'background': {
+                                        'image_link': '/base/static/img/preview_background.png',
+                                        'preview_link': '/base/static/pdf/preview_background.pdf'
+                                    },
+                                    'boxed': {
+                                        'image_link': '/base/static/img/preview_boxed.png',
+                                        'preview_link': '/base/static/pdf/preview_boxed.pdf'
+                                    },
+                                    'clean': {
+                                        'image_link': '/base/static/img/preview_clean.png',
+                                        'preview_link': '/base/static/pdf/preview_clean.pdf'
+                                    },
+                                    'standard': {
+                                        'image_link': '/base/static/img/preview_standard.png',
+                                        'preview_link': '/base/static/pdf/preview_standard.pdf'
+                                    }
+                                }" />
+                        <field name="report_header"  widget="char"
+                               placeholder="e.g. Global Business Solutions" />
+                        <field name="report_footer" string="Footer"
+                               placeholder="e.g. Opening hours, bank accounts (one per line)" />
+                        <field name="paperformat_id" widget="selection" />
                     </group>
                     <footer>
-                        <button special="save" string="Save" class="btn btn-primary" type="object" />
+                        <button special="save" string="Save"
+                                class="btn btn-primary" type="object" />
                         <button special="cancel" string="Cancel" />
                     </footer>
                 </form>
